### PR TITLE
fix: [computer]Getting the UUID is stuck, causing the interface to freeze

### PR DESCRIPTION
**Pull request by pengfeixx**
Put the obtained UUID into the thread for execution

Log: Fixed some known issues
Bug: https://pms.uniontech.com/bug-view-254659.html

### CODE REVIEW

File: src/dfm-base/base/device/deviceproxymanager.cpp [hunk 1]
```diff
@@ -345,6 +374,29 @@ void DeviceProxyManagerPrivate::disconnCurrentConnections()
...
+QVariantMap DeviceProxyManagerPrivate::asyncQueryInfo(const QString &id, bool reload, std::function<QDBusPendingReply<QVariantMap>(const QString&, bool)> func)
+{
+    QEventLoop loop;
+    QFutureWatcher<QDBusPendingReply<QVariantMap>>* fw = new QFutureWatcher<QDBusPendingReply<QVariantMap>>;
+    connect(fw, &QFutureWatcher<QDBusPendingReply<QVariantMap>>::finished, [&loop](){
+        loop.quit();
+    });
+    QFuture<QDBusPendingReply<QVariantMap>> ft =
+    QtConcurrent::run([id, reload, func](){
+        DeviceManagerInterface devMngDBus(kDeviceService, kDevMngPath, QDBusConnection::sessionBus());
```
Comment (itsXuSt) [inline src/dfm-base/base/device/deviceproxymanager.cpp]: unused

File: src/dfm-base/base/device/deviceproxymanager.cpp [hunk 2]
```diff
@@ -80,6 +83,32 @@ QVariantMap DeviceProxyManager::queryProtocolInfo(const QString &id, bool reload
...
+QVariantMap DeviceProxyManager::asyncQueryBlockInfo(const QString &id, bool reload)
+{
+    if (d->isDBusRuning() && d->devMngDBus) {
+        auto fun = [=](const QString &id, bool reload) -> QDBusPendingReply<QVariantMap> {
+            DeviceManagerInterface devMngDBus(kDeviceService, kDevMngPath, QDBusConnection::sessionBus());
+            return devMngDBus.QueryBlockDeviceInfo(id, reload);
+        };
+        return d->asyncQueryInfo(id, reload, fun);
+    } else {
+        return queryBlockInfo(id, reload);
```
Comment (itsXuSt) [inline src/dfm-base/base/device/deviceproxymanager.cpp:108]: 你这个函数产生二义性了。这里依然是同步操作。

**Review by itsXuSt — CHANGES_REQUESTED** [review]

File: src/dfm-base/base/device/deviceproxymanager.cpp [hunk 3]
```diff
@@ -40,7 +41,9 @@ QStringList DeviceProxyManager::getAllBlockIdsByUUID(const QStringList &uuids, G
     const auto &&devices = getAllBlockIds(opts);
     QStringList devs;
     for (const auto &id : devices) {
-        const auto &&info = queryBlockInfo(id);
+        const auto &&info = asyncQueryBlockInfo(id);
```
Comment (Johnson-zs) [inline src/dfm-base/base/device/deviceproxymanager.cpp:57]: 同步接口中调用异步接口？

File: src/dfm-base/base/device/deviceproxymanager.h [hunk 4]
```diff
@@ -36,8 +36,10 @@ class DeviceProxyManager : public QObject
     QStringList getAllBlockIds(GlobalServerDefines::DeviceQueryOptions opts = GlobalServerDefines::DeviceQueryOption::kNoCondition);
     QStringList getAllBlockIdsByUUID(const QStringList &uuids, GlobalServerDefines::DeviceQueryOptions opts = GlobalServerDefines::DeviceQueryOption::kNoCondition);
     QStringList getAllProtocolIds();
-    QVariantMap queryBlockInfo(const QString &id, bool reload = false);
-    QVariantMap queryProtocolInfo(const QString &id, bool reload = false);
+    QVariantMap queryBlockInfo(const QString &id, bool reload = false, bool async = false);
```
Comment (Johnson-zs) [inline src/dfm-base/base/device/deviceproxymanager.h]: 确定能编译过么

**Review by Johnson-zs — CHANGES_REQUESTED** [review]

File: src/dfm-base/base/device/deviceproxymanager.cpp (hunk 1 above)
Comment (pengfeixx) [reply src/dfm-base/base/device/deviceproxymanager.cpp]: > unused

已修改

File: src/dfm-base/base/device/deviceproxymanager.h (hunk 4 above)
Comment (pengfeixx) [reply src/dfm-base/base/device/deviceproxymanager.h]: > 确定能编译过么

已修改

**Review by Johnson-zs — APPROVED** [review]

**Pull request merged**